interface/backend/cases/models.py

```diff
@@ -140,18 +146,36 @@ class Candidate(models.Model):
     probability_concerning = models.FloatField(validators=[MinValueValidator(0.0), MaxValueValidator(1.0)])
     review_result = models.IntegerField(choices=enums.format_enum(enums.CandidateReviewResult),
                                         default=enums.CandidateReviewResult.NONE)
+    added_by_hand = models.BooleanField(default=False)
+
+    def get_or_create_nodule(self):
+        nodule, created = Nodule.objects.get_or_create(candidate=self)
+        return nodule, created
+
+    def remove_associated_nodule(self):
+        return Nodule.objects.filter(candidate=self).delete()
 
 
 class Nodule(models.Model):
     """
-    Actual nodule, either confirmed as concerning from prediction or manually added.
+    Actual nodule, originating from a candidates which was marked as concerning
     """
     created = models.DateTimeField(default=timezone.now)
-    case = models.ForeignKey(Case, on_delete=models.CASCADE, related_name='nodules')
     candidate = models.OneToOneField(Candidate, on_delete=models.CASCADE, null=True)
-    centroid = models.OneToOneField('images.ImageLocation', on_delete=models.CASCADE)
     lung_orientation = models.IntegerField(choices=enums.format_enum(enums.LungOrientation),
                                            default=enums.LungOrientation.NONE)
     appearance_feature = models.IntegerField(choices=enums.format_enum(enums.AppearanceFeature), null=True)
     diameter = models.FloatField(null=True)
     density_feature = models.IntegerField(choices=enums.format_enum(enums.DensityFeature), null=True)
+
+
+@receiver(post_save, sender=Candidate)
+def add_or_remove_nodule_once_candidate_reviewed(sender, instance, *args, **kwargs):
```

Comment:
Can we put these into the ``save`` methods? Signals are really magic-at-a-distance with weird error handling semantics :(